### DISCUSSION TITLE
SSO Functionality?

### DISCUSSION THREAD

**Discussion by jasonpierson** [oidc-wp/openid-connect-generic#305] [Q&A]
I am in a situation similar to the person here: https://github.com/oidc-wp/openid-connect-generic/issues/70
My site has information that you do not need a login for, but also features that you do need a login for. Has the plugin been updated so there a way for this plugin to "perceive" that the user is logged in via SSO and then automatically log them in on the site?

Since the issue was marked stale and closed, I'm guessing not. If that's the case, is there a good, free openid plugin that does this? I came across this one: https://github.com/dirtsimple/oidc-sso
Has anyone used it? Is it any good?

**Comment by timnolte** [comment]
@jasonpierson so are you wanting the WordPress site to proactively, without any user request to "login", check if there is already an active Keycloak session for the current user visiting the site? Other users(https://github.com/oidc-wp/openid-connect-generic/issues/285) are using Keycloak to authenticate with other platforms for provide protected content.

**Comment by jasonpierson** [comment]
@timnolte, yes, I am wanting the WordPress site to proactively, without any user request to "login", check if there is already an active Keycloak session for the current user visiting the site and log them in if so.
